src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js

```diff
@@ -0,0 +1,113 @@
+import processRadioSchedule, {
+  getLink,
+  getProgramState,
+} from './processRadioSchedule';
+import persianRadioScheduleData from '#data/persian/bbc_persian_radio/schedule.json';
+
+describe('getLink', () => {
+  let program;
+  let service;
+
+  beforeAll(() => {
+    service = 'persian';
+    program = {
+      ...persianRadioScheduleData.schedules[0],
+      serviceId: 'bbc_dari_radio',
+      episode: { pid: 'p07zbtbf' },
+    };
+  });
+  it('should return liveradio link when state is live', () => {
+    expect(getLink('live', program, service)).toBe(
+      '/persian/bbc_dari_radio/liveradio',
+    );
+  });
+  it('should return program link when state is not live', () => {
+    expect(getLink('anyotherstate', program, service)).toBe(
+      '/persian/bbc_dari_radio/p07zbtbf',
+    );
+  });
+});
+
+describe('getProgramState', () => {
+  it('should return `live` when currentTime is greater than startTime but less than endTime', () => {
+    const currentTime = Date.now();
+    const startTime = Date.now() - 1000;
+    const endTime = Date.now() + 1000;
```

Comment:
Thanks for all of these unit tests.

I'd recommend updating all these tests to use the currentTime and increment that. The reason is, that from the time when the first Date.now() is run to the next time Date.now() is run, time elapses, so they won't have the same identical reference point.
```suggestion
    const startTime = currentTime - 1000;
    const endTime = currentTime + 1000;
```

Reply:
Thanks @sareh . Those tests have now been updated